template.yml.erb

```diff
@@ -41,42 +41,153 @@ Globals:
 Conditions:
   IsDevCondition: !Equals [!Ref BaseDomainName, "dev-code.org"]
 Resources:
-  Domain:
+# Note: We can't update the name of a DomainName resource once it has been created because the
+# domain name itself has already been provisioned. When we change from javabuilderbeta to
+# javabuilder, we should update the WebSocket resources here to use the prefix "WebSocket"
+<%{
+  Http: {Prefix: "Http", Suffix: "-http"},
+  WebSocket: {Prefix: "", Suffix: ""},
+}.each do |apiName, config| -%>
+  <%=config[:Prefix]%>Domain:
     Type: AWS::Route53::RecordSet
     Properties:
       HostedZoneName: !Sub "${BaseDomainName}."
-      Name: !Sub "${SubDomainName}.${BaseDomainName}"
+      Name: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
       Type: A
       AliasTarget:
-        DNSName: !GetAtt DomainName.RegionalDomainName
-        HostedZoneId: !GetAtt DomainName.RegionalHostedZoneId
+        DNSName: !GetAtt <%=config[:Prefix]%>DomainName.RegionalDomainName
+        HostedZoneId: !GetAtt <%=config[:Prefix]%>DomainName.RegionalHostedZoneId
 
-  DomainName:
+  <%=config[:Prefix]%>DomainName:
     Type: AWS::ApiGatewayV2::DomainName
     Properties:
-      DomainName: !Sub "${SubDomainName}.${BaseDomainName}"
+      DomainName: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
       DomainNameConfigurations:
         - EndpointType: REGIONAL
-          CertificateArn: !Ref Certificate
-          CertificateName: !Sub "${SubDomainName}.${BaseDomainName}"
+          CertificateArn: !Ref <%=config[:Prefix]%>Certificate
+          CertificateName: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
 
-  Certificate:
+  <%=config[:Prefix]%>Certificate:
     Type: AWS::CertificateManager::Certificate
     Properties:
-      DomainName: !Sub "${SubDomainName}.${BaseDomainName}"
+      DomainName: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
       ValidationMethod: DNS
       DomainValidationOptions:
-        - DomainName: !Sub "${SubDomainName}.${BaseDomainName}"
+        - DomainName: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
           HostedZoneId: !Ref BaseDomainNameHostedZonedID
 
-  DomainNameAPIMapping:
+  <%=config[:Prefix]%>DomainNameAPIMapping:
     Type: AWS::ApiGatewayV2::ApiMapping
     DependsOn:
-      - Domain
+      - <%=config[:Prefix]%>Domain
     Properties:
-      ApiId: !Ref WebSocketAPI
-      DomainName: !Sub "${SubDomainName}.${BaseDomainName}"
-      Stage: !Ref WebSocketStage
+      ApiId: !Ref <%=apiName%>API
+      DomainName: !Sub "${SubDomainName}<%=config[:Suffix]%>.${BaseDomainName}"
+      Stage: !Ref <%=apiName%>Stage
+
+<%end -%>
+  HttpAPI:
+    Type: AWS::ApiGatewayV2::Api
+    Properties:
+      Name: !Sub "${SubDomainName}-http.${BaseDomainName}"
+      ProtocolType: HTTP
+
+  PutRoute:
+    Type: AWS::ApiGatewayV2::Route
+    Properties:
+      ApiId: !Ref HttpAPI
+      RouteKey: PUT /seedsources/sources.json
```

Comment:
Probably too late to change this, but I would think the Resource we're trying to PUT is the source code of a project, so the path of this API would be something like `/project/sources.json` and eventually we'll implement the PUT `/project/assets/whatever.jpg` or `/project/assets/assets.zip` routes.